src/main/java/com/smartcar/sdk/SmartcarException.java

```diff
@@ -98,107 +33,57 @@ public SmartcarException() {
+   * Returns the message associated with the exception.
    *
-   * @param request the associated HTTP request
-   * @param response the associated HTTP response
+   * @return the message
    */
-  public SmartcarException(final Request request, final Response response) {
-    this(null, Status.forCode(response.code()), request, response);
+  public String getMessage() {
+    return this.message;
   }
 
   /**
-   * Initializes a new Smartcar API exception with the specified message,
-   * status, request, and response.
+   * Returns the error code associated with the exception.
    *
-   * @param message a message associated with the exception
-   * @param status a status associated with the exception
-   * @param request the associated HTTP request
-   * @param response the associated HTTP response
+   * @return the error string code
    */
-  public SmartcarException(final String message, final Status status, final Request request, final Response response) {
-    super(SmartcarException.generateMessage(message, status));
-
-    if(status != null) {
-      this.status = status;
-    }
-    else {
-      this.status = Status.UNKNOWN;
-    }
-
-    this.request = request;
-    this.response = response;
+  public String getError() {
+    return this.error;
   }
 
   /**
-   * Generates a message for the exception, handling cases where no message is provided.
+   * Returns the HTTP status code.
    *
-   * @param message the message specified when the exception was thrown
-   * @param status the status specified when the exception was thrown
-   *
-   * @return the generated message
+   * @return the response status code
    */
-  private static String generateMessage(final String message, final Status status) {
-    if(message != null) {
-      return message;
-    }
-
-    if(status != null) {
-      return status.toString();
-    }
-
-    return Status.UNKNOWN.toString();
+  public Integer getStatusCode() {
+    return this.statusCode;
   }
 
-  /**
-   * Returns the status associated with the SmartcarException.
-   *
-   * @return the associated status
-   */
-  public Status getStatus() {
-    return this.status;
-  }
-
-  /**
-   * Returns the HTTP request associated with the exception.
-   *
-   * @return the request object
-   */
-  public Request getRequest() {
-    return this.request;
-  }
+  public static SmartcarException Factory(final Response response) throws IOException {
+    JsonObject body = gson.fromJson(response.body().string(), JsonObject.class);
 
-  /**
-   * Returns the HTTP response associated with the exception.
-   *
-   * @return the response object
-   */
-  public Response getResponse() {
-    return this.response;
+    String code = "";
+    JsonElement codeElement = body.get("code");
+    if (codeElement != null) {
+      code = codeElement.getAsString();
+    }
+    String message = body.get("message").getAsString();
+    String error = body.get("error").getAsString();
+    Integer statusCode = response.code();
```

Comment:
Why `Integer` and not `int`? Is that what this returns?

Reply:
should be `int`, done~